### PR TITLE
Inline module info question #59

### LINKED ISSUE #59: Inline module info question
Would it be possible to inline the options of module info in the Java project wizard page?

image

All the other options are set on the first page, only the module info is done via an additional popup which feels weird and IMHO is bad UX.

### PR DESCRIPTION
**Pull request by ktatavarthi**
Inline module info question #59. Fixes #59

Signed-off-by: Kalyan Prasad Tatavarthi <kalyan_prasad@in.ibm.com>

### CODE REVIEW

File: org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/wizards/NewJavaProjectWizardPageOne.java [hunk 1]
```diff
@@ -1440,6 +1579,31 @@ public boolean getCreateModuleInfoFile() {
...
+	 * Returns the module name given.
+	 *
+	 * @return 'Module name:' text field value, removed of all spaces. If empty , returns
+	 * the project name, removed of all spaces.
+	 * @since 3.27
+	 */
+	public String getModuleName() {
+		String moduleName = fModuleGroup.getModuleName();
+		if (moduleName == null || moduleName.length() == 0) {
+			moduleName= fNameGroup.getName().replaceAll(WHITE_SPACE_STR, EMPTY_STR);
```
Comment (SarikaSinha) [inline org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/wizards/NewJavaProjectWizardPageOne.java:1592]: Do we need to do this or just throw invalid error like we do in Create module-info dialog?

Comment (SarikaSinha) [reply org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/wizards/NewJavaProjectWizardPageOne.java]: Looks good now.

File: org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/wizards/NewJavaProjectWizardPageOne.java [hunk 2]
```diff
@@ -804,17 +811,31 @@ public IWorkingSet[] getSelectedWorkingSets() {
 		}
 	}
 
-	private final class ModuleGroup implements IDialogFieldListener{
+	private final class ModuleGroup extends Observable implements IDialogFieldListener{
```
[the remark this answers is not in the document]
Comment (SarikaSinha) [reply org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/wizards/NewJavaProjectWizardPageOne.java:814]: Looks good now.

**Comment by SarikaSinha** [comment]
Looks good.

**Comment by SarikaSinha** [comment]
Not part of this change, but looking at the code noticed that should the source and compliance be same as derived now or should be retrieved from Java Project like this: 
javaProject.getOption(JavaCore.COMPILER_SOURCE, true)
javaProject.getOption(JavaCore.COMPILER_COMPLIANCE, true)

**Comment by ktatavarthi** [comment]
> Not part of this change, but looking at the code noticed that should the source and compliance be same as derived now or should be retrieved from Java Project like this: javaProject.getOption(JavaCore.COMPILER_SOURCE, true) javaProject.getOption(JavaCore.COMPILER_COMPLIANCE, true)

We cannot use this because we are creating the java project here.

**Pull request merged**